contracts/RewardPool/RewardPool.sol

```diff
@@ -79,15 +79,18 @@ contract RewardPool is RewardPoolBase, System, StakingRewards, DelegationRewards
     ) private returns (uint256 reward) {
         require(uptime.signedBlocks <= totalBlocks, "SIGNED_BLOCKS_EXCEEDS_TOTAL");
 
-        (, , uint256 totalStake, uint256 commission, , ) = validatorSet.getValidator(uptime.validator);
+        (, , uint256 totalStake, uint256 commission, , bool active) = validatorSet.getValidator(uptime.validator);
+        if (!active) {
+            return 0;
```

Comment:
Validator may receive its reward for a previous epoch when he has been still active